### PR TITLE
Configuration guide

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: docs/installation_configuration_guide/ModelConfiguration.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+  campaignType: {
+    editor: { type: "StringParameterEditor" }
+    validators: [ { type: "MandatoryParameterValidator" } ]
+    label: "Campaign type"
+    defaultValue: "Generic campaign"
+  }
+  ...
+}
+```
+Configure is similar to [component configuration](#configuration-of-ui-attributes-of-components)
```
Comment (dswiecki) [inline docs/installation_configuration_guide/ModelConfiguration.md:176]:
```suggestion
Configuration is similar to [component configuration](#configuration-of-ui-attributes-of-components)
```

File: docs/installation_configuration_guide/DeploymentManagerConfiguration.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+```
+deploymentConfig {     
+  type: "flinkStreaming"
+  restUrl: "http://localhost:8081"
+}
+```
+
+`flinkStreaming` deployment manager has following configuration options:
+
+| Parameter | Type | Default value | Description  |
```
Comment (dswiecki) [inline docs/installation_configuration_guide/DeploymentManagerConfiguration.md:14]: pipe format

File: docs/installation_configuration_guide/Configuration.md [hunk 3]
```diff
@@ -0,0 +1 @@
+
```
Comment (dswiecki) [inline docs/installation_configuration_guide/Configuration.md:1]: I understand we will put content here later on right ?

File: docs/installation_configuration_guide/DesignerConfiguration.md [hunk 4]
```diff
@@ -0,0 +1,604 @@
...
+| db.minimumIdle       | Low        | int    | 1                                                         | We have lower limits than default config, since then designer is not heavy-load application |
+| db.numThreads        | Low        | int    | 5                                                         | We have lower limits than default config, since then designer is not heavy-load application |
+
+## Metrics settings
+                                                                     
+### Metric dashboard
+
+Each scenario can have link to grafana dashboard. In [docker setup](https://github.com/TouK/nussknacker/tree/staging/demo/docker/grafana) we 
+provide `nussknacker-scenario` dashboard. 
+You can modify/configure own, the only assumption that we make is that [variable](https://grafana.com/docs/grafana/latest/variables/) is used to display metrics for particular scenario.
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/installation_configuration_guide/DesignerConfiguration.md]: done, a bit differently

File: docs/installation_configuration_guide/DesignerConfiguration.md [hunk 5]
```diff
@@ -0,0 +1,604 @@
...
+  }
+]
+```
+
+#### Encrypted hashes
+
+Encryption of passwords uses BCrypt algorithm. You can generate sample hash using command:
+
+```
+python -c 'import bcrypt; print(bcrypt.hashpw("password".encode("utf8"), bcrypt.gensalt(rounds=12, prefix="2a")))'
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/installation_configuration_guide/DesignerConfiguration.md:182]: :+1: added note

File: docs/installation_configuration_guide/DesignerConfiguration.md [hunk 6]
```diff
@@ -0,0 +1,604 @@
...
+
+When `method` is set to `OAuth2`, the following fields are mandatory: `clientSecret`, `clientId`, `authorizeUri`
+, `redirectUri`, `accessTokenUri`, `profileUri`, `profileFormat`, `implicitGrantEnabled`, `usersFile`.
+
+For the `profileFormat` one of `oidc` or `github` is supported by default.
+
+Subconfigs `accessTokenParams`, `accessTokenRequestContentType`, `authorizeParams`, `headers` are optional and every
+field from any of the subconfigs is optional and could be provided separately.
+
+By default access token request is sent using `application/json` content type, to use change it (eg.
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/installation_configuration_guide/DesignerConfiguration.md]: :+1:  corrected

File: docs/installation_configuration_guide/ModelConfiguration.md [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+| checkpointConfig.minPauseBetweenCheckpoints        | Low        | duration | checkpointInterval / 2 | [Minimal pause](https://ci.apache.org/projects/flink/flink-docs-stable/docs/deployment/config/#execution-checkpointing-min-pause) between checkpoints                       |
+| checkpointConfig.maxConcurrentCheckpoints          | Low        | int      | 1                      | [Maximum concurrent checkpoints](https://ci.apache.org/projects/flink/flink-docs-stable/docs/deployment/config/#execution-checkpointing-max-concurrent-checkpoints) setting |
+| checkpointConfig.tolerableCheckpointFailureNumber  | Low        | int      |                        | [Tolerable failed checkpoint](https://ci.apache.org/projects/flink/flink-docs-stable/docs/deployment/config/#execution-checkpointing-tolerable-failed-checkpoints) setting  |
+| rocksDB.checkpointDataUri                          | High       | string   |                        | URI for                                                                                                                                                                     |
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/installation_configuration_guide/ModelConfiguration.md]: :+1: I forgot about this one...

File: docs/installation_configuration_guide/ModelConfiguration.md [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+## Kafka configuration
+
+Currently, it's only possible to use one Kafka cluster for one model configuration. This configuration is used for all
+Kafka based sources and sinks
+                      
+Important thing to remember is that Kafka server addresses/schema registry addresses have to be resolvable from:
+- Nussknacker Designer host (to enable schema discovery and scenario testing)
+- Flink cluster (both jobmanagers and taskmanagers) hosts - to be able to run job
+
+| Name                                                                              | Importance | Type                       | Default value    | Description  |
```
[the remark this answers is not in the document]
Comment (mproch) [reply docs/installation_configuration_guide/ModelConfiguration.md]: done

File: docs/installation_configuration_guide/DeploymentManagerConfiguration.md (hunk 2 above)
Comment (mproch) [reply docs/installation_configuration_guide/DeploymentManagerConfiguration.md:14]: done

File: docs/installation_configuration_guide/Configuration.md (hunk 3 above)
Comment (mproch) [reply docs/installation_configuration_guide/Configuration.md:1]: I added main part

**Review by dswiecki — APPROVED** [review]

**Pull request merged**